### PR TITLE
Wazo 3259 connect user timeout

### PR DESCRIPTION
**Pull request by DrPyser**
https://wazo-dev.atlassian.net/browse/WAZO-3259

Depends-On: https://github.com/wazo-platform/wazo-calld-client/pull/55

### CODE REVIEW

**Comment by DrPyser** [comment]
recheck

**Comment by DrPyser** [comment]
depends-on: https://github.com/wazo-platform/wazo-calld-client/pull/55

**Comment by DrPyser** [comment]
recheck

File: integration_tests/suite/test_calls.py [hunk 1]
```diff
@@ -2305,6 +2305,52 @@ def test_given_one_call_and_one_user_when_connect_user_then_the_two_are_talking(
...
+    def test_one_call_one_user_with_timeout(
+        self,
+    ):
+        call_id = new_call_id()
+        my_new_call_id = new_call_id(leap=1)
+        self.ari.set_channels(
+            MockChannel(id=call_id),
+            MockChannel(id=my_new_call_id),
+        )
+        self.ari.set_channel_variable({my_new_call_id: {'XIVO_USERUUID': 'user-uuid'}})
```
Comment (sduthil) [inline integration_tests/suite/test_calls.py:2338]: Please use `WAZO_USERUUID` where possible. `XIVO_USERUUID` is deprecated.

File: wazo_calld/plugins/calls/http.py [hunk 2]
```diff
@@ -274,11 +276,25 @@ def put(self, call_id):
...
+            body = connect_call_request_body_schema.load(request.json)
+
+        args = {
+            'call_id': call_id,
+            'user_uuid': user_uuid,
+        }
+
+        if body:
+            # NOTE(clanglois): setting the 'timeout' to null
+            # allows disabling the timeout(infinite ringing)
```
Comment (sduthil) [inline wazo_calld/plugins/calls/http.py:294]: This important info should go in the documentation, not in a comment.

File: wazo_calld/plugins/calls/services.py [hunk 3]
```diff
@@ -328,7 +328,7 @@ def hangup_user(self, call_id, user_uuid):
         self._verify_user(call_id, user_uuid)
         self._ari.channels.hangup(channelId=call_id)
 
-    def connect_user(self, call_id, user_uuid):
+    def connect_user(self, call_id, user_uuid, timeout=30):
```
Comment (sduthil) [inline wazo_calld/plugins/calls/services.py:331]: The default value of 30 seconds should only be defined once. It is currently defined twice: here and in the schema.

**Review by sduthil — CHANGES_REQUESTED** [review]
Thanks Charles!

A few other things are missing:
* update the API doc
* update the API changelog
* Test when timeout is invalid

File: wazo_calld/plugins/calls/http.py (hunk 2 above)
Comment (DrPyser) [reply wazo_calld/plugins/calls/http.py:294]: why not both?
Agreed, I'll add it to the openapi doc.

**Review by sduthil — APPROVED** [review]

**Review by sduthil — APPROVED** [review]

**Pull request merged**